WordPress/src/main/java/org/wordpress/android/util/NetworkUtilsWrapper.kt

```diff
@@ -0,0 +1,13 @@
+package org.wordpress.android.util
+
+import org.wordpress.android.WordPress
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class NetworkUtilsWrapper @Inject constructor() {
```

Comment:
Let's pick an approach between this approach and [ConnectionStatusLiveData](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/viewmodel/helpers/ConnectionStatusLiveData.kt) and use it everywhere. It's not great to add 2 different components that achieves the same thing immediately after one another.

I am fine with either approach, so feel free to go with whichever you prefer, but if you do want to use this one, it'd be great if you can open a PR to remove the `ConnectionStatusLiveData`.

Also, can you please update the base branch with `develop`? Our tests and reviews would be more reliable if we can keep up with the base branch. If you go with the above change, it'd be necessary to do so anyway.

Reply:
>>Let's pick an approach between this approach and ConnectionStatusLiveData and use it everywhere. It's not great to add 2 different components that achieves the same thing immediately after one another.

>Hmm, I don't consider them to be the same components. One registers a broadcast receiver which can be a bit of an overkill and the second one just asks on the current status. However, I'm fine with keeping just a single component. The ConnectionStatusLiveData is more powerful so it's an easy choice.

Ok, so it's not so easy choice. I think we should keep both of them. IMHO they serve a different purpose.
The `ConnectionStatusLiveData` needs to be observed in order to register to the Broadcast -> we'd have to start observing in the view layer and propagate the changes to the VM. However, since we want to start fetching data in `vm.onStart()` we can't be sure the broadcast reciever is already active and we'll receive the result in time. Moreover, it feels like we are using something aimed for a different purpose.... I think the `ConnectionStatusLiveData` is useful for cases where we want to for example automatically update the UI when the connection status changes. Which is not the case. Wdyt?